numpyro/handlers.py

```diff
@@ -221,7 +221,7 @@ def process_message(self, msg):
             if guide_msg["type"] != "sample" or guide_msg["is_observed"]:
                 raise RuntimeError(f"Site {name} must be sampled in trace.")
             msg["value"] = guide_msg["value"]
-            msg["infer"] = guide_msg["infer"]
+            msg["infer"] = guide_msg["infer"].copy()
```

Comment:
@martinjankowiak @fehiepsi  Is it okay to copy the `infer` dictionary? Can that possibly mess up other inference algorithms?

I had to create a copy here because computing `dim_to_name` for the site in the model overwrites it in the guide as well. But there can be cases like in new tests in this PR when the guide's `dim_to_name` is different than the one in the model.

Reply:
I think it is fine.